capstone/fitness/src/main/java/com/google/sps/servlets/AbstractAuthServlet.java

```diff
@@ -1,42 +1,150 @@
 package com.google.sps.servlets;
-
+ 
 import com.google.appengine.api.users.UserService;
 import com.google.appengine.api.users.UserServiceFactory;
-
 import java.io.IOException;
-
-
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import javax.servlet.ServletException;
-
-
 import com.google.api.client.extensions.appengine.auth.oauth2.AbstractAppEngineAuthorizationCodeServlet;
 import com.google.api.client.auth.oauth2.AuthorizationCodeFlow;
+import com.google.sps.util.*;
+import com.google.api.services.calendar.Calendar;
+import com.google.api.client.util.DateTime;
+import com.google.api.services.calendar.model.Event;
+import com.google.api.services.calendar.model.Events;
+import java.util.*;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.client.extensions.appengine.http.UrlFetchTransport;
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.services.calendar.model.CalendarList;
+import com.google.gson.Gson;
+import com.google.api.services.calendar.model.CalendarListEntry;
+import java.time.format.DateTimeFormatter;  
+import java.time.LocalDateTime;   
+import com.google.sps.util.*;
+import com.google.sps.progress.*;
+import com.google.gson.reflect.TypeToken;
+
 
-@WebServlet("/abstract")
+
+// AbstractAuthServlet initializes the OAuth process. 
+@WebServlet("/auth-servlet")
 public class AbstractAuthServlet extends AbstractAppEngineAuthorizationCodeServlet {
   String nickname = UserServiceFactory.getUserService().getCurrentUser().getNickname();
+  String APPLICATION_NAME = "Marathon App";
+  Calendar calendar;
+  Scheduler scheduler;
+  long exerciseDuration = 30;  
+  Gson gson = new Gson();
+  DateTimeFormatter dtf = DateTimeFormatter.ofPattern("MM/dd/YYYY");  
+  LocalDateTime now = LocalDateTime.now();  
 
+  
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    System.out.println("Nickname " + nickname);
-    response.setContentType("text/html;");
-    response.getWriter().println(nickname + " has authorized this app.");
-  }
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
```

Comment:
This method is very long and does a couple of different things. Can you think about pulling out logic into separate methods for easier readability?

Reply:
Yes!